### PR TITLE
Use same task definition for deploy_service and update_service

### PR DESCRIPTION
**Pull request by aswinkarthik**

### CODE REVIEW

File: cloudlift/deployment/deployer.py [hunk 1]
```diff
@@ -53,26 +68,36 @@ def deploy_task_definition(client, task_definition, cluster_name, ecs_service_na
...
-    if complete_image_uri is not None:
-        task_definition.set_images(essential_container, deploy_version_tag, **{essential_container: complete_image_uri})
-    else:
-        task_definition.set_images(essential_container, deploy_version_tag)
-    for container in task_definition.containers:
-        env_config = container_configurations.get(container['name'], {})
-        task_definition.apply_container_environment_and_secrets(container, env_config)
-        task_definition.apply_memory_hard_limit(HARD_LIMIT_MEMORY_IN_MB)
+    task_definition.compute_diffs(essential_container, ecr_image_uri)
     print_task_diff(ecs_service_name, task_definition.diff, color)
```
Comment (siliconsenthil) [inline cloudlift/deployment/deployer.py:81]: Do we need this, given you are finding diff and printing all params?
```    essential_container = find_essential_container(task_definition[u'containerDefinitions'])
    container_configurations = build_config(env_name, service_name, ecs_service_logical_name, sample_env_file_path,
                                            essential_container,
                                            secrets_name)
    task_definition.compute_diffs(essential_container, ecr_image_uri)
    print_task_diff(ecs_service_name, task_definition.diff, color)```

File: cloudlift/deployment/task_definition_builder.py [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+
+
+class TaskDefinitionBuilder:
+    def __init__(self, environment, service_name, configuration, region):
+        self.environment = environment
+        self.service_name = service_name
+        self.configuration = configuration
+        self.region = region
+
+    def build_dict(self,
```
Comment (siliconsenthil) [inline cloudlift/deployment/task_definition_builder.py:20]: Let's make this private? Unit tests can still call pvt methods.

File: cloudlift/deployment/task_definition_builder.py [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+                                                                    {})),
+                )
+        if launch_type == LAUNCH_TYPE_FARGATE:
+            td_kwargs['RequiresCompatibilities'] = [LAUNCH_TYPE_FARGATE]
+            td_kwargs['NetworkMode'] = 'awsvpc'
+            td_kwargs['Cpu'] = str(config['fargate']['cpu'])
+            td_kwargs['Memory'] = str(config['fargate']['memory'])
+        else:
+            if 'memory' in config:
+                td_kwargs['Memory'] = str(config.get('memory'))
```
Comment (siliconsenthil) [inline cloudlift/deployment/task_definition_builder.py:164]: Do we need task-defn level memory?

File: cloudlift/deployment/task_definition_builder.py [hunk 4]
```diff
@@ -0,0 +1,233 @@
...
+            Environment=[Environment(Name=k, Value=v) for (k, v) in env_config],
+            MemoryReservation=int(sidecar.get('memory_reservation')),
+            Image=sidecar.get('image'),
+            LogConfiguration=log_config,
+            Essential=False,
+            **cd
+        )
+
+
+def camelize_keys(data):
```
Comment (siliconsenthil) [inline cloudlift/deployment/task_definition_builder.py:204]: - This method does more than just camelizing key names. It should renamed accordingly or split into two methods.
- we should add unit test
- Make this private?

File: cloudlift/deployment/task_definition_builder.py [hunk 5]
```diff
@@ -0,0 +1,233 @@
...
+            if v == 'true' or v == 'false':
+                result[key] = v == 'true'
+            else:
+                result[key] = v
+        else:
+            result[key] = v
+    return result
+
+
+def _camel_case(value):
```
Comment (siliconsenthil) [inline cloudlift/deployment/task_definition_builder.py:228]: Use the library

File: cloudlift/deployment/deployer.py [hunk 6]
```diff
@@ -53,26 +68,36 @@ def deploy_task_definition(client, task_definition, cluster_name, ecs_service_na
...
+        container_configurations=container_configurations,
+        ecr_image_uri=ecr_image_uri,
+        fallback_task_role=task_definition.role_arn,
+        fallback_task_execution_role=task_definition.execution_role_arn,
+    ))
+    diff = DeepDiff(task_definition, updated_task_definition)
+    diff.pop('dictionary_item_removed', 'no dictionary_item_removed')
+
+    if diff:
+        log_intent(pformat(diff))
```
Comment (siliconsenthil) [inline cloudlift/deployment/deployer.py:99]: Can you paste how it appears here?

Comment (aswinkarthik) [reply cloudlift/deployment/deployer.py:99]: Its a JSON
``` json
 {'values_changed': {"root['containerDefinitions'][0]['image']": {'new_value': 'nginx:v2',
                                                                 'old_value': 'nginx:v1'}}}
```

File: cloudlift/deployment/task_definition_builder.py (hunk 3 above)
Comment (aswinkarthik) [reply cloudlift/deployment/task_definition_builder.py:164]: We don't need it. I removed it.

File: cloudlift/deployment/task_definition_builder.py (hunk 4 above)
Comment (aswinkarthik) [reply cloudlift/deployment/task_definition_builder.py:204]: Renamed it to `_cloudformation_to_boto3_payload`

File: cloudlift/deployment/task_definition_builder.py (hunk 2 above)
Comment (aswinkarthik) [reply cloudlift/deployment/task_definition_builder.py:20]: Renamed it to `build_task_definition` 
It is a public method accessed by `deployer.py`

File: cloudlift/deployment/task_definition_builder.py (hunk 5 above)
Comment (aswinkarthik) [reply cloudlift/deployment/task_definition_builder.py:228]: Using `stringcase.camelcase` now

File: cloudlift/deployment/deployer.py (hunk 1 above)
Comment (aswinkarthik) [reply cloudlift/deployment/deployer.py:81]: This still prints the old text
Old Image  was "v1" and new image is "v2". I thought we can retain that. The other diff is a raw diff with a JSON output. We may not be able to process all the changes for all the fields.

**Review by siliconsenthil — APPROVED** [review]

**Pull request merged**